### PR TITLE
Fixed #3381 - Workflow field gets id instead of value

### LINKED ISSUE #3381: Workflow field gets id instead of value
Create a task that subject equals account name in workflow and get id instead of value 
#### Issue
Shows wrong value when create task in opportunities module 

#### Expected Behavior
When you want the subject to be the account name it is the account name and not it's id.

#### Actual Behavior
When you create a workflow with the subject = account name in the opportunities module.

#### Steps to Reproduce

1. In new install  with 7.8.2
2. create workflow in opportunities module
3.
image

4. get id instead of text in created  task
image

#### Context
Unable to create proper workflow for opportunities module
Feel it is High Priority 

#### Your Environment

Version  7.8.2
Sugar Version 6.5.24 (Build 509)
Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
Safari/602.1 but also test with Chrome/opera/ firefox (all the latest versions)
Environment name and version (e.g. MySQL, PHP 7):
MySQL PHP6.5
Operating System and version (e.g Ubuntu 16.04):
GNU/Linux

### PR DESCRIPTION
**Pull request by ChangezKhan**
## Description

It fixes issue #3381 

Create a task that subject equals account name in workflow and get id instead of value

Steps to reproduce:
1.  Create workflow in opportunities module
2.  Create new task on create of opportunity.
3.  Set task's subject to account name.
4. Save and create new opportunity with any account record.

## Motivation and Context

It creates a task that subject equals account name in workflow.

## How To Test This

1.  Create workflow in opportunities module
2.  Create new task on create of opportunity.
3.  Set task's subject to account name.
4. Save and create new opportunity with any account record.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/AOW_Actions/actions/actionCreateRecord.php [hunk 1]
```diff
@@ -170,12 +170,14 @@ function set_record(SugarBean $record, SugarBean $bean, $params = array(), $in_s
                             case 'int':
                                 $value = format_number($bean->$fieldName);
                                 break;
-                            case 'relate':
-                                if(isset($data['id_name'])) {
-                                    $idName = $data['id_name'];
+			    case 'relate':
+			        if(isset($data['id_name']) && $record_vardefs[$field]['type'] == 'relate'){
```
Comment (daniel-samson) [inline modules/AOW_Actions/actions/actionCreateRecord.php]: You can safely use ===. Please can you change the == to ===?

**Comment by daniel-samson** [comment]
@Dillon-Brown can you mark this as in review?

File: modules/AOW_Actions/actions/actionCreateRecord.php (hunk 1 above)
Comment (ChangezKhan) [reply modules/AOW_Actions/actions/actionCreateRecord.php]: So sorry for late response. I have changed code as per you requested.

**Comment by Dillon-Brown** [comment]
Assessed

**Pull request merged**